Packs/RecordedFuture/Integrations/RecordedFuture/RecordedFuture.py

```diff
@@ -1357,6 +1371,122 @@ def __build_links_context(self, links_data: Dict[str, List], markdown: str) -> C
                     outputs_key_field=""
                 )
 
+    def fetch_incidents(self, rule_names_arg: Optional[str], fetch_time: str) -> None:
+        if rule_names_arg:
+            rule_names = rule_names_arg.split(';')
+        else:
+            rule_names = []
+
+        if not fetch_time:
+            fetch_time = '24 hours'
+
+
+        last_run = demisto.getLastRun()
+        if not last_run:
+            last_run = {}
+        if 'time' not in last_run:
+            time, _ = parse_date_range(fetch_time, date_format='%Y-%m-%dT%H:%M:%S.%fZ')
+        else:
+            time = last_run['time']
+
+        current_time = datetime.strptime(time, '%Y-%m-%dT%H:%M:%S.%fZ')
+        triggered_time = '[{},)'.format(datetime.strftime(current_time, '%Y-%m-%d %H:%M:%S'))
+        max_time = current_time
+
+        rule_ids = []  # type: list
+
+        for rule in rule_names:
+            rules = self.client.get_alert_rules(rule)
+            if rules and 'data' in rules:
```

Comment:
Not sure I understand this if statement.
Can you please explain?

Reply:
there might be a response but with `error` key and no `data` key

Reply:
Thanks.
Is `rules` a string/dictionary ?

Reply:
it's a dictionary.